### ISSUE TITLE
eslint complains about `global-require` even if the require usage is scoped

### ISSUE THREAD

**Issue by satazor** [eslint/eslint#4812]
```js
const someFunc = require('./someFunc');

someFunc(function (require) {
    require('bananas', function () {  // eslint complains that I should put this require at the top

   });
});
```

Obviously one could rename `require` to `req` to make this issue go away, but I think eslint should be smarter and not complain in the situation exemplified above.

Thoughts?

Version: 1.10.3

**Comment by eslintbot** [comment]
@satazor Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
Agreed. We should be able to easily make sure reported require calls are referencing the global require function.

**Comment by nzakas** [comment]
This is working as intended. The goal of the rule is to mimic how ES6 module imports work.

Your best bet in this situation is to disable the rule around that usage.

**Comment by michaelficarra** [comment]
@nzakas You may want to re-read the issue. The rule is flagging calls to *any* function named `require`.

**Comment by satazor** [comment]
@nzakas I don't think it is intended because the `require` is a function argument, not a global require.

**Comment by nzakas** [comment]
Thanks @michaelficarra @btmills and @ilyavolodin for pointing out that I misunderstood this issue. I was too quick to close it.

@satazor you're right, we should be able to tell the difference between a reference to the global `require` and a different one.

**Comment by platinumazure** [comment]
I could use some experience with escope. Anyone mind if I take this on this week?

**Comment by nzakas** [comment]
Go for it

**Comment by Willibaur** [comment]
Hi all, I have a similar situation than @satazor here is my piece of code where ESlin is complaining
```javascript
  const moment = require('moment');
  function formatDate(date) {
    return moment(date).format('MMMM Do YYYY');
  }
```
According to the DOCS that is completely valid code,
```javascript
// requiring a module and using it in a function is ok
var fs = require('fs');
function readFile(filename, callback) {
    fs.readFile(filename, callback)
}
```
So pls let me know what am I doing wrong here.
Thanks in advance.

**Comment by btmills** [comment]
@Willibaur if you haven't figured out what was going on yet, can you open a new issue and @-mention me? The information in our bug report template will help us track down what's happening. For what it's worth, the code looks correct to me, and the [demo](http://eslint.org/demo) doesn't complain about it.
